debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -598,6 +602,61 @@ public Object getColumnValue(ResultSet rs, int columnIndex, Column column, Table
         }
     }
 
+    // NOTE: fix for DBZ-7359
+    @Override
+    public void setQueryColumnValue(PreparedStatement statement, Column column, int pos, Object value) throws SQLException {
+        boolean isColumnValueSet = false;
```

Comment:
I think instead of the flag you can intorduce `default` branch in the `switch` and `return` for the conditional

Reply:
I removed the `isColumnValueSet` flag but the logic with the flag always set the value, so I had to add several additional else statements to get the equivalent logic. I'm not terribly keen on either approach as they are both brittle to maintenance changes. Thoughts?